tokio/src/runtime/thread_pool/worker.rs

```diff
@@ -605,56 +623,31 @@ where
         tick
     }
 
-    fn is_searching(&mut self) -> bool {
+    fn is_searching(&self) -> bool {
         self.owned().is_searching.get()
     }
 
-    fn set(&self) -> &Set<P> {
-        &self.pool
+    fn index(&self) -> usize {
+        self.worker.index
     }
 
-    fn shared(&self) -> &Shared<P> {
-        &self.set().shared()[self.index]
+    fn slices(&self) -> &slice::Set<P::Unpark> {
+        &self.worker.slices
     }
 
-    fn owned(&mut self) -> &Owned<P> {
-        // safety: we own the slot
-        unsafe { &*self.set().owned()[self.index].get() }
+    fn shared(&self) -> &Shared<P::Unpark> {
+        &self.slices().shared()[self.index()]
     }
-}
 
-struct DropNotGone<'a, T> {
```

Comment:
Not needed anymore as `Park` is stored in an `Arc` w/ `generation` as a mutual exclusion guard.